### PR TITLE
add swapping and bridging

### PR DESCRIPTION
**Pull request by rodrigopavezi**

### CODE REVIEW

File: src/components/Donations/DonationShelf/index.tsx [hunk 1]
```diff
@@ -29,22 +29,18 @@ const DonationShelf = () => {
...
+        return <ProcessCrypto {...{ setStage, donationMethod, order, amount, setOrder }} />
       case 'confirmation':
         return <Confirmation />
       case 'setAmount':
       default:
         return <DonationAmount {...{ setStage, setDonationMethod, donationMethod, setAmount, amount }} />
     }
   }
-  return renderSwitch(stage, donationMethod)
+  return <WagmiConfig client={client}>{renderSwitch(stage, donationMethod)}</WagmiConfig>
```
Comment (EricWVGG) [inline src/components/Donations/DonationShelf/index.tsx:43]: could we move this up to wrap `<ProcessCrypto>` instead?

or even better, could we move it to ProcessCrypto.tsx and then `export default <WagmiConfig><ProcessCrypto /></WagmeConfig>` ?

File: src/components/Donations/PaymentCrypto/index.tsx [hunk 2]
```diff
@@ -78,7 +92,47 @@ const PaymentCrypto = ({ setStage, amount, donationMethod, chains, setOrder }: I
 
   const processTransaction = async () => {
     if (donationMethod === 'polygon') {
-      write?.()
+      transferUSDC?.()
```
Comment (EricWVGG) [inline src/components/Donations/PaymentCrypto/index.tsx:95]: oh cool, I didn't know you could call an optional method with a period like that…

File: src/components/Donations/ProcessCrypto/index.tsx [hunk 3]
```diff
@@ -40,10 +60,23 @@ const ProcessCrypto = ({ setStage, donationMethod, order }: IProcessCrypto) => {
...
-      if (message.title === 'Payment is COMPLETE') {
-        handleTopUpComplete()
+      switch (message.title) {
+        case 'Payment is COMPLETE':
+          handleTopUpComplete()
+          break
+        case 'Swap is COMPLETE':
+          handleSwapComplete()
+          break
+        default:
```
Comment (EricWVGG) [inline src/components/Donations/ProcessCrypto/index.tsx:78]: might be a good idea to `throw new Error(`unknown response: ${message.title})` or `console.alert(message.title)`

**Review by EricWVGG — CHANGES_REQUESTED** [review]

File: src/components/Donations/DonationShelf/index.tsx (hunk 1 above)
Comment (rodrigopavezi) [reply src/components/Donations/DonationShelf/index.tsx:43]: I need it on both ProcessCrypto and PaymentCrypto that is why I moved it up

File: src/components/Donations/ProcessCrypto/index.tsx (hunk 3 above)
Comment (rodrigopavezi) [reply src/components/Donations/ProcessCrypto/index.tsx:78]: Added it

Comment (rodrigopavezi) [reply src/components/Donations/ProcessCrypto/index.tsx:78]: Yeap

**Review by EricWVGG — APPROVED** [review]

**Pull request merged**